### ISSUE TITLE
The local vendor/bin proxy scripts are incorrect when built using a junction on a different drive on Windows.

### ISSUE THREAD

**Issue by rquadling** [composer/composer#5428]
This issue came about through https://github.com/composer/composer/issues/5417#issuecomment-225224604

TL;DR; The vendor\bin proxy files for repos imported via a Windows Junction do not have the correct local path. See the end of this issue.

L;

I've recreated the issue using an unadulterated installation of Rob Morgan's `Phinx`.

Requires a windows installation with 2 drives (C: and D: for example).

Main repo is installed in C:\test and has the following `composer.json`

```json
{
    "name": "rquadling/test",
    "description": "test",
    "license": "proprietary",
    "repositories": [
        {
            "type": "path",
            "url": "D:/Sources/PHP/RobMorgan-Phinx"
        }
    ],
    "require" : {
    	"robmorgan/phinx" : "dev-0.5.x-dev"
    }
}
```

In D:\Source\PHP\RobMorgan-Phinx, I have a clone of the Rob Morgan' Phinx.

When I run this command:

```
composer install -vvv -o -a
```

I get this output:

```
[ 13/06/2016 10:55:06 ] [ C:\test ] >composer install -vvv -o -a
Reading ./composer.json
Loading config file C:/Users/RichardQ/AppData/Roaming/Composer/config.json
Loading config file C:/Users/RichardQ/AppData/Roaming/Composer/auth.json
Loading config file ./composer.json
Checked CA file C:\Users\RichardQ\AppData\Local\Temp\composer-cacert-bcf331364fef73d53e4368ba519a6c3fcb21ea7836c3d8efad4ec32731e8152a.pem: valid
Executing command (C:\test): git branch --no-color --no-abbrev -v
Executing command (C:\test): git describe --exact-match --tags
Reading C:/Users/RichardQ/AppData/Roaming/Composer/composer.json
Loading config file C:/Users/RichardQ/AppData/Roaming/Composer/config.json
Loading config file C:/Users/RichardQ/AppData/Roaming/Composer/auth.json
Loading config file C:/Users/RichardQ/AppData/Roaming/Composer/composer.json
Loading config file C:\Users\RichardQ\AppData\Roaming\Composer/auth.json
Reading C:/Users/RichardQ/AppData/Roaming/Composer/vendor/composer/installed.json
Running 1.1.2 (2016-05-31 19:48:11) with PHP 5.4.45 on Windows NT / 6.2
Loading composer repositories with package information
Executing command (D:\Sources\PHP\RobMorgan-Phinx\): git branch --no-color --no-abbrev -v
Executing command (D:\Sources\PHP\RobMorgan-Phinx\): git log -n1 --pretty=%H
Downloading https://packagist.org/packages.json
Writing C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2013.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2014.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015-07.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015-10.json from cache
Downloading http://packagist.org/p/provider-2016-01%24d75b881a45e9414d8eacd14447f7c24841f699291a4eddc8001855b4c3dde3b0.json
Writing C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2016-01.json into cache
Downloading http://packagist.org/p/provider-2016-04%24fe37785c8fa5a8b20dd245416b319d2aab487754eb69b64af42b3287736cecf5.json
Writing C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-2016-04.json into cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-archived.json from cache
Downloading http://packagist.org/p/provider-latest%24f9e58c05c909a6436daae8b94e33ebacb8e2495e2487437903fc5a0883db6be2.json
Writing C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/p-provider-latest.json into cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-robmorgan$phinx.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$console.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$config.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$yaml.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-mbstring.json from cache
Reading C:/Users/RichardQ/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$filesystem.json from cache
Resolving dependencies through SAT
Dependency resolution completed in 0.032 seconds
Analyzed 957 packages to resolve dependencies
Analyzed 7020 rules to resolve dependencies
  - Installing symfony/yaml (v2.8.7)
Reading C:/Users/RichardQ/AppData/Local/Composer/files/symfony/yaml/465bcd6e9dbae388c6025c68ac4d3228da238082.zip from cache
    Loading from cache
    Extracting archive

    REASON: robmorgan/phinx dev-0.5.x-dev requires symfony/yaml ~2.8|~3.0 -> satisfiable by symfony/yaml[v2.8.0, v2.8.1, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.1.0].

  - Installing symfony/filesystem (v2.8.7)
Reading C:/Users/RichardQ/AppData/Local/Composer/files/symfony/filesystem/4847eb1dcc73356dda00b31abb1517500e1b7ab9.zip from cache
    Loading from cache
    Extracting archive

    REASON: symfony/config v2.8.7 requires symfony/filesystem ~2.3|~3.0.0 -> satisfiable by symfony/filesystem[v2.3.0, v2.3.1, v2.3.10, v2.3.11, v2.3.12, v2.3.13, v2.3.14, v2.3.15, v2.3.16, v2.3.17, v2.3.18, v2.3.19, v2.3.2, v2.3.20, v2.3.21, v2.3.22, v2.3.23, v2.3.24, v2.3.25, v2.3.26, v2.3.27, v2.3.28, v2.3.29, v2.3.3, v2.3.30, v2.3.31, v2.3.32, v2.3.33, v2.3.34, v2.3.35, v2.3.36, v2.3.37, v2.3.38, v2.3.39, v2.3.4, v2.3.40, v2.3.41, v2.3.42, v2.3.5, v2.3.6, v2.3.7, v2.3.8, v2.3.9, v2.4.0, v2.4.1, v2.4.10, v2.4.2, v2.4.3, v2.4.4, v2.4.5, v2.4.6, v2.4.7, v2.4.8, v2.4.9, v2.5.0, v2.5.1, v2.5.10, v2.5.11, v2.5.12, v2.5.2, v2.5.3, v2.5.4, v2.5.5, v2.5.6, v2.5.7, v2.5.8, v2.5.9, v2.6.0, v2.6.1, v2.6.10, v2.6.11, v2.6.12, v2.6.13, v2.6.2, v2.6.3, v2.6.4, v2.6.5, v2.6.6, v2.6.7, v2.6.8, v2.6.9, v2.7.0, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.2, v2.7.3, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7].

  - Installing symfony/config (v2.8.7)
Reading C:/Users/RichardQ/AppData/Local/Composer/files/symfony/config/d1fe3b9beb011a767c79e56fe0be8cab3d3286c4.zip from cache
    Loading from cache
    Extracting archive

    REASON: robmorgan/phinx dev-0.5.x-dev requires symfony/config ~2.8|~3.0 -> satisfiable by symfony/config[v2.8.0, v2.8.1, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.1.0].

  - Installing symfony/polyfill-mbstring (v1.2.0)
Reading C:/Users/RichardQ/AppData/Local/Composer/files/symfony/polyfill-mbstring/24d40f06ac52ce041e58490f228cff12f192e0a2.zip from cache
    Loading from cache
    Extracting archive

    REASON: symfony/console v2.8.7 requires symfony/polyfill-mbstring ~1.0 -> satisfiable by symfony/polyfill-mbstring[v1.0.0, v1.0.1, v1.1.0, v1.1.1, v1.2.0].

  - Installing symfony/console (v2.8.7)
Reading C:/Users/RichardQ/AppData/Local/Composer/files/symfony/console/43f8df54c259ba1e21f17fb4651a2cd72685689d.zip from cache
    Loading from cache
    Extracting archive

    REASON: robmorgan/phinx dev-0.5.x-dev requires symfony/console ~2.8|~3.0 -> satisfiable by symfony/console[v2.8.0, v2.8.1, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.1.0].

  - Installing robmorgan/phinx (dev-0.5.x-dev)
    Junctioned from D:/Sources/PHP/RobMorgan-Phinx

    REASON: Required by the root package: Install command rule (install robmorgan/phinx dev-0.5.x-dev|install robmorgan/phinx dev-0.5.x-dev)

symfony/console suggests installing symfony/event-dispatcher ()
symfony/console suggests installing symfony/process ()
symfony/console suggests installing psr/log (For using the console logger)
Reading C:\test/vendor/composer/installed.json
Writing lock file
Generating optimized autoload files
```

On the surface, this is all appropriate output.

Note that the `phinx` installation has been via a junction:

```
  - Installing robmorgan/phinx (dev-0.5.x-dev)
    Junctioned from D:/Sources/PHP/RobMorgan-Phinx
```

Doing some simple analysis of the installation reveals all files are in the appropriate place and the classmap is aware of them.

```
[ 13/06/2016 10:56:46 ] [ C:\test ] >type vendor\composer\autoload_classmap.php | grep -i phinx
    'Phinx\\Config\\Config' => $vendorDir . '/robmorgan/phinx/src/Phinx/Config/Config.php',
    'Phinx\\Config\\ConfigInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Config/ConfigInterface.php',
    'Phinx\\Console\\Command\\AbstractCommand' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/AbstractCommand.php',
    'Phinx\\Console\\Command\\Create' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Create.php',
    'Phinx\\Console\\Command\\Init' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Init.php',
    'Phinx\\Console\\Command\\Migrate' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Migrate.php',
    'Phinx\\Console\\Command\\Rollback' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Rollback.php',
    'Phinx\\Console\\Command\\SeedCreate' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/SeedCreate.php',
    'Phinx\\Console\\Command\\SeedRun' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/SeedRun.php',
    'Phinx\\Console\\Command\\Status' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Status.php',
    'Phinx\\Console\\Command\\Test' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/Command/Test.php',
    'Phinx\\Console\\PhinxApplication' => $vendorDir . '/robmorgan/phinx/src/Phinx/Console/PhinxApplication.php',
    'Phinx\\Db\\Adapter\\AdapterFactory' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/AdapterFactory.php',
    'Phinx\\Db\\Adapter\\AdapterInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/AdapterInterface.php',
    'Phinx\\Db\\Adapter\\AdapterWrapper' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/AdapterWrapper.php',
    'Phinx\\Db\\Adapter\\MysqlAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/MysqlAdapter.php',
    'Phinx\\Db\\Adapter\\PdoAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/PdoAdapter.php',
    'Phinx\\Db\\Adapter\\PostgresAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/PostgresAdapter.php',
    'Phinx\\Db\\Adapter\\ProxyAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/ProxyAdapter.php',
    'Phinx\\Db\\Adapter\\SQLiteAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/SQLiteAdapter.php',
    'Phinx\\Db\\Adapter\\SqlServerAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/SqlServerAdapter.php',
    'Phinx\\Db\\Adapter\\TablePrefixAdapter' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/TablePrefixAdapter.php',
    'Phinx\\Db\\Adapter\\WrapperInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Adapter/WrapperInterface.php',
    'Phinx\\Db\\Table' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Table.php',
    'Phinx\\Db\\Table\\Column' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Table/Column.php',
    'Phinx\\Db\\Table\\ForeignKey' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Table/ForeignKey.php',
    'Phinx\\Db\\Table\\Index' => $vendorDir . '/robmorgan/phinx/src/Phinx/Db/Table/Index.php',
    'Phinx\\Migration\\AbstractMigration' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/AbstractMigration.php',
    'Phinx\\Migration\\CreationInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/CreationInterface.php',
    'Phinx\\Migration\\IrreversibleMigrationException' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/IrreversibleMigrationException.php',
    'Phinx\\Migration\\Manager' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/Manager.php',
    'Phinx\\Migration\\Manager\\Environment' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/Manager/Environment.php',
    'Phinx\\Migration\\MigrationInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Migration/MigrationInterface.php',
    'Phinx\\Seed\\AbstractSeed' => $vendorDir . '/robmorgan/phinx/src/Phinx/Seed/AbstractSeed.php',
    'Phinx\\Seed\\SeedInterface' => $vendorDir . '/robmorgan/phinx/src/Phinx/Seed/SeedInterface.php',
    'Phinx\\Util\\Util' => $vendorDir . '/robmorgan/phinx/src/Phinx/Util/Util.php',
    'Phinx\\Wrapper\\TextWrapper' => $vendorDir . '/robmorgan/phinx/src/Phinx/Wrapper/TextWrapper.php',
```

Show the junction via windows
```
[ 13/06/2016 10:57:17 ] [ C:\test ] >dir vendor\robmorgan
 Volume in drive C is Windows7
 Volume Serial Number is 46D9-C513

 Directory of C:\test\vendor\robmorgan

13/06/2016  10:55    <DIR>          .
13/06/2016  10:55    <DIR>          ..
13/06/2016  10:55    <JUNCTION>     phinx [D:\Sources\PHP\RobMorgan-Phinx]
               0 File(s)              0 bytes
               3 Dir(s)  759,639,535,616 bytes free
```

Show the junction/link via Cygwin
```
[ 13/06/2016 10:57:21 ] [ C:\test ] >ls vendor/robmorgan -al
total 0
drwxrwx---+ 1 Administrators RichardQ  0 Jun 13 10:55 .
drwxrwx---+ 1 Administrators RichardQ  0 Jun 13 10:55 ..
lrwxrwxrwx  1 Administrators RichardQ 39 Jun 13 10:55 phinx -> /cygdrive/d/Sources/PHP/RobMorgan-Phinx
```

Examining the `C:\test\vendor\bin` directory
```
[ 13/06/2016 10:57:47 ] [ C:\test ] >dir vendor\bin
 Volume in drive C is Windows7
 Volume Serial Number is 46D9-C513

 Directory of C:\test\vendor\bin

13/06/2016  10:55    <DIR>          .
13/06/2016  10:55    <DIR>          ..
13/06/2016  10:55               617 phinx
13/06/2016  10:55               131 phinx.bat
               2 File(s)            748 bytes
               2 Dir(s)  759,638,745,088 bytes free
```

Shows the 2 bin proxy scripts.

vendor\bin\phinx
```sh
#!/usr/bin/env sh

dir=$(d=${0%[/\\]*}; cd "$d"; cd "d:/Sources/PHP/RobMorgan-Phinx/bin" && pwd)

# See if we are running in Cygwin by checking for cygpath program
if command -v 'cygpath' >/dev/null 2>&1; then
        # Cygwin paths start with /cygdrive/ which will break windows PHP,
        # so we need to translate the dir path to windows format. However
        # we could be using cygwin PHP which does not require this, so we
        # test if the path to PHP starts with /cygdrive/ rather than /usr/bin
        if [[ $(which php) == /cygdrive/* ]]; then
                dir=$(cygpath -m "$dir");
        fi
fi

dir=$(echo $dir | sed 's/ /\ /g')
"${dir}/phinx" "$@"
```

and vendor\bin\phinx.bat
```bat
@ECHO OFF
setlocal DISABLEDELAYEDEXPANSION
SET BIN_TARGET=%~dp0/d:/Sources/PHP/RobMorgan-Phinx/bin/phinx
php "%BIN_TARGET%" %*
```

The line
```
SET BIN_TARGET=%~dp0/d:/Sources/PHP/RobMorgan-Phinx/bin/phinx
```
is incorrect.

If changed to ...
```
SET BIN_TARGET=%~dp0/../robmorgan/phinx/bin/phinx
```

all works OK.

Strangely enough, the unix shell script works without amendments from within Cygwin, but I do not know if this is correct.

If changed to 
```
dir=$(d=${0%[/\\]*}; cd "$d"; cd "../robmorgan/phinx/bin" && pwd)
```
the script still works but is operating via the junction and not on the physical location.

**Comment by rquadling** [comment]
No need having 2 issues.
